sdk/appconfiguration/app-configuration/types/app-configuration.d.ts

```diff
@@ -0,0 +1,387 @@
+     * filtered by key names, labels and accept datetime.
+     *
+     * Example code:
+     * ```ts
+     * const allSettingsWithLabel = await client.listConfigurationSettings({ labels: [ "MyLabel" ] });
+     * ```
+     * @param options Optional parameters for the request.
+     */
+    listConfigurationSettings(options?: ListConfigurationSettingsOptions): PagedAsyncIterableIterator<ConfigurationSetting, ListConfigurationSettingPage>;
+    private getListConfigurationSettingsIterator;
+    private listConfigurationSettingsByPage;
+    private createListConfigurationPageFromResponse;
+    /**
+     * Lists revisions of a set of keys, optionally filtered by key names,
+     * labels and accept datetime.
+     *
+     * Example code:
+     * ```ts
+     * const revisionsIterator = await client.listRevisions({ keys: ["MyKey"] });
+     * ```
+     * @param options Optional parameters for the request.
+     */
+    listRevisions(options?: ListRevisionsOptions): PagedAsyncIterableIterator<ConfigurationSetting, ListRevisionsPage>;
+    private getListRevisionsIterator;
+    private listRevisionsByPage;
+    /**
+     * Sets the value of a key in the Azure App Configuration service, allowing for an optional etag.
+     * @param key The name of the key.
+     * @param configurationSetting A configuration value.
+     * @param options Optional parameters for the request.
+     *
+     * Example code:
+     * ```ts
+     * await client.setConfigurationSetting({ key: "MyKey", value: "MyValue" });
+     * ```
+     */
+    setConfigurationSetting(configurationSetting: SetConfigurationSettingParam, options?: SetConfigurationSettingOptions): Promise<SetConfigurationSettingResponse>;
+    /**
+     * Sets a key's value to read only
+     * @param id The id of the configuration setting to set to read-only.
+     */
+    setReadOnly(id: ConfigurationSettingId, options?: SetReadOnlyOptions): Promise<SetReadOnlyResponse>;
+    /**
+     * Makes the key's value writable again
+     * @param id The id of the configuration setting to make writable.
+     */
+    clearReadOnly(id: ConfigurationSettingId, options?: ClearReadOnlyOptions): Promise<ClearReadOnlyResponse>;
+}
+
+/**
+ * Options for clearReadOnly
+ */
+export declare interface ClearReadOnlyOptions extends HttpOnlyIfUnchangedField, RequestOptionsBase {
+}
+
+/**
+ * Response when clearing the read-only status from a value
+ */
+export declare interface ClearReadOnlyResponse extends ConfigurationSetting, SyncTokenHeaderField, HttpResponseField<SyncTokenHeaderField> {
+}
+
+/**
+ * Configuration setting with extra metadata from the server, indicating
+ * its etag, whether it is currently readOnly and when it was last modified.
+ */
+export declare interface ConfigurationSetting extends ConfigurationSettingParam {
+    /**
+     * Whether or not the setting is read-only
+     */
+    readOnly: boolean;
+    /**
+     * The date when this setting was last modified
+     */
+    lastModified?: Date;
```

Comment:
`lastModifiedOn`?

Reply:
Yes.

Reply:
Consensus was to keep it consistent with the HTTP header (and other languages) - so stay as 'lastModified'